package/lib/user/user_repository.dart

```diff
@@ -0,0 +1,61 @@
+import 'dart:async';
+import 'dart:convert';
+
+import '../constants.dart';
+import '../entity/domain.dart';
+import '../utils/http.dart';
+import 'user_entity.dart';
+
+final class UserData extends UserEntity {
+  UserData._internal();
+
+  static final userData = UserData._internal();
+
+  factory UserData() => userData;
+}
+
+class UserRepository {
+  final _userData = UserData();
+
+  UserEntity get data {
+    return _userData;
+  }
+
+  void set(UserEntity user) {
+    _userData.userID = user.userID;
+    if (user.cpf != null) _userData.cpf = user.cpf;
+    if (user.name != null) _userData.name = user.name;
+    if (user.email != null) _userData.email = user.email;
+    if (user.gender != null) _userData.gender = user.gender;
+    if (user.birthday != null) _userData.birthday = user.birthday;
+    if (user.location != null) _userData.location = user.location;
+    if (user.customData != null) _userData.customData = user.customData;
+  }
+
+  FutureOr<bool> identify(UserEntity? user) async {
+    if (user != null) set(user);
```

Comment:
O que você acha da gente só exportar esse identify, ter o _set como função interna, e sempre chamar ele aqui? Assim, unificamos as funções de identify, setUser, identifyUser, chamando só uma vez o identify, e as outras não precisam fazer nenhuma manipulação.

Reply:
devido a formato anterior, precisamos ter o set. O identify chama o set e o identifyUser chama o identify. Sendo assim se for método privado o identify antigo vai dar problema.

Reply:
Então, a minha ideia era o identify chamar o identify do repository, e o identifyUser não fazer mais nada. Daria certo?

Reply:
não devido ao cenário atual. O dev primeiro faz um set (pelo identify) e depois chama o identifyUser. Não podemos garantir que vai ter todas as informações necessárias no momento do identify, e como não temos o update do usuário, teremos problemas com dados inconsistentes.